### DISCUSSION TITLE
Do you have a relay server milestone?

### DISCUSSION THREAD

**Discussion by dohankk** [WalletConnect/walletconnect-monorepo#3772] [Developer Support] [answered]
In relay repository, Docs say `Self-hosting is currently not supported and WalletConnect will not provide support for implementations relying on self-hosting until announced otherwise.`

Is there a time limit on when a relay will be created?

**Comment by glitch-txs** [comment, marked as answer]
We are working on decentralization, no ETA at the moment.

**Reply by tfalencar** [reply]
I find it dishonest to use the term decentralization while at the same time keeping your running server code closed. This is not a pragmatic approach towards decentralization: decentralized > self-hosting/federated > centralized. From wallet connect V1 to V2 we went from the self-hosting/federated, to fully centralized (today).

Your v2 server code **is still closed source**, which makes **everyone**  use your **centralized** backend service (since v1 was officially deprecated).

**Reply by glitch-txs** [reply]
Hi, we'll open source the relay on Q1 early next year

**Reply by renestefancic** [reply]
@glitch-txs any update on this?

**Reply by sunopar** [reply]
@glitch-txs any update?

**Reply by arein** [reply]
@tfalencar @renestefancic @sunopar we have a wide set of node operators now on the Relay. It's not fully permissionless yet. But WalletConnect are on the way to becoming permissionless. Please visit https://walletconnect.network/ and follow WalletConnect on Twitter for updates
